### ISSUE TITLE
Light Client: Iteration on map storage will never stop.

### ISSUE THREAD

**Issue by david9991** [paritytech/subxt#1525]
The storage `devices` in my testnet has only one entry, but the iteration will never stop while querying via the light client.
Dependences:
```
subxt = { version = "0.35.2", features = ["unstable-light-client", "unstable-reconnecting-rpc-client"] }
subxt-signer = { version = "0.35.2", features = ["subxt"] }
subxt-lightclient = "0.35.2"
```

<img width="1573" alt="Screenshot 2024-04-12 at 15 18 34" src="https://github.com/paritytech/subxt/assets/123219/ffa77e2e-0cf3-45b4-bf41-edc35de39f32">

**Comment by jsdw** [comment]
Thanks for your report! Oh that is strange; def something we'll need to dig into!

Would you happen to have some concrete steps to reproduce this? Does it happen when iterating over any storage query or just a specific one? Does it still happen when you're _not_ using the reconnecting RPC client?

**Comment by david9991** [comment]
@jsdw You can reproduce it using this project: https://github.com/david9991/lightclient-lssue

**Comment by lexnv** [comment]
IIRC, we have encountered something similar with the legacy backend.

Could you change the backend to use the unstable one (rpc-v2 unstable methods), as suggested here? 🙏 
https://github.com/paritytech/subxt/issues/1453#issuecomment-2011922808

**Comment by jsdw** [comment]
@david9991 Thank you for your reproduciton; that was very helpful!

The problem is that the Smoldot storage API we use returns the `start_key` given whereas the Substrate one does not. As a workaround, I created a small fix in #1534 which will ignore such a dupe key if returned. This makes your reproduction work as expected :)
